db/migrations/2011222206_mark_results_with_answers_and_hints_as_started.sql

```diff
@@ -0,0 +1,58 @@
+-- +migrate Up
+
+/* Set started_at for results having at least one of latest_submission_at or latest_hint_at set */
+UPDATE `results`
+SET`started_at` = COALESCE(
+    (SELECT MIN(`created_at`) FROM `answers`
+    WHERE `answers`.`participant_id` = `results`.`participant_id` AND
+          `answers`.`attempt_id` = `results`.`attempt_id` AND
+          `answers`.`item_id` = `results`.`item_id`),
+    `latest_submission_at`, `latest_hint_at`),
+    started_at = LEAST(`started_at`,
+        IFNULL(`latest_submission_at`, `latest_hint_at`),
+        IFNULL(`latest_hint_at`, `latest_submission_at`))
+WHERE (`latest_submission_at` IS NOT NULL OR `latest_hint_at` IS NOT NULL) AND `started_at` IS NULL;
+
+/*
+  Set started_at for results having descendant results with started_at set for the same attempt.
+  Items linked to both ancestor and descendant results must be visible for the participant.
```

Comment:
I don't get this sentence. There are still visible even if not started. 🤔

Reply:
We do not start a parent result if the child result is started, but the parent (or child) item is not visible.

Reply:
I don't get your answer.
But no worries, you can merge (we can discuss on slack)

Reply:
fixed